lib/streamlit/runtime/scriptrunner/script_run_context.py

```diff
@@ -60,9 +61,9 @@ class ScriptRunContext:
     session_state: SafeSessionState
     uploaded_file_mgr: UploadedFileManager
     main_script_path: str
-    page_script_hash: str
     user_info: UserInfo
     fragment_storage: "FragmentStorage"
+    pages_manager: "PagesManager"
```

Comment:
curious: when do we use the TYPE_CHECKING only typing versus the other? To me it seems to be a little bit arbitrary

Reply:
My interpretation is that the TYPE_CHECKING is used for imports and to avoid circular dependencies.